api/v1alpha1/clusterextension_types.go

```diff
@@ -183,6 +183,9 @@ type ClusterExtensionStatus struct {
 //+kubebuilder:object:root=true
 //+kubebuilder:resource:scope=Cluster
 //+kubebuilder:subresource:status
+//+kubebuilder:printcolumn:name="Installed Version",type=string,JSONPath=`.status.installedBundle.version`
```

Comment:
is it possible to float up any of the conditions or errors? As a user it would be good to understand: where the CR is in its progression, version state (as above), and some kind of sense of where it stopped (and possibly why?)

Reply:
I think it is possible to write JSON path to do something like "find condition with type "Installed" and tell me it's reason", but that might not be all that helpful because:
1. Showing details of every condition type would get pretty verbose
2. Only showing details of _some_ condition types could be misleading.

I wonder if, after @bentito's work to wrangle condition updates to reflect reality better, we could do something where right before we send out the status update we pass the conditions into a function that returns $something that somewhat summarizes the state of the conditions. For example:
- Installed=False, Progressing=False => "Failed"
- Installed=False, Progressing=True => "Installing"
- Installed=True, Progressing=False, Healthy=False => "Unhealthy"
- Installed=True, Progressing=False, Healthy=True => "Ready"
- Installed=True, Progressing=True => "Upgrading"
- etc.

Reply:
That could be cool. It would just be nice to get some signal without having to inspect the whole status.

Reply:
I've been looking into this, and it appears the JSON is limited to being very simply, and you really can't do something that complicated with another field having that value.

Reply:
Error I got when I tried to be a bit more "advanced":
```
The CustomResourceDefinition "clusterextensions.olm.operatorframework.io" is invalid: spec.additionalPrinterColumns[4].JSONPath: Invalid value: "{range .status.conditions[?(@.type=='Installed' && @.status=='False')]}{range .status.conditions[?(@.type=='Progressing' && @.status=='True')]}{'Installed'}{end}{end}": must be a simple json path starting with .